srtcore/sync.h

```diff
@@ -943,6 +943,81 @@ CUDTException& GetThreadLocalError();
 /// @param[in] maxVal maximum allowed value of the resulting random number.
 int genRandomInt(int minVal, int maxVal);
 
+class SharedMutex
+{
+    private:
+    Condition  m_LockWriteCond;
+    Condition  m_LockReadCond;
+
+    Mutex m_Mutex;
+    int m_iCountRead;
+    bool m_bWriterLocked;
```

Comment:
These two fields can be combined into one field. Please see `/usr/include/c++/v1/shared_mutex`.

Would be also nice that you use exactly the same public API as the standard one so that a prospective drop-in replacement with the standard one from C++17 can be done later.

Reply:
I thought about merging those two fields. I however felt that it would hurt readibility for a negligible memory and performance gain. 

API has been changed to match the standard one